server/src/main/java/org/candlepin/policy/js/compliance/hash/HashableStringGenerators.java

```diff
@@ -211,7 +211,15 @@ public String generate(Consumer target) {
             }
 
             String generated = target.getUuid();
-            generated += generateFromCollection(target.getFacts().entrySet(), STRING_ENTRY);
+
+            if (target.getFacts() != null) {
```

Comment:
The ComplianceReasonGenerator has the same potential issue with generating from attributes. We should probably fix that as well.

Reply:
I traced back the code. It seems highly unlikely that we might run into NPE as it create a new instance of ```ComplianceStatus``` which has a [constructor](https://github.com/candlepin/candlepin/blob/master/server/src/main/java/org/candlepin/policy/js/compliance/ComplianceStatus.java#L59) that initialize ```ComplianceReason```. Do you have any particular scenario where we could run into such situations ?